src/testing/core/mocking/__tests__/mockLink.ts

```diff
@@ -1,7 +1,55 @@
 import gql from "graphql-tag";
-import { MockLink } from "../mockLink";
+import { MockLink, MockedResponse } from "../mockLink";
 import { execute } from "../../../../link/core/execute";
 
+/*
+These are Type check tests for ensuring MockedResponse.newData type works as expected
+*/
+// Untyped MockedResponse means newData is weakly typed
+const passingTesForNewDataUntypedMockedResponse: MockedResponse = {
+  request: {
+    query: gql`query A { a }`,
+  },
+
+  newData: ({ fake: { faker }}) => ({
+    data: {
+      pretend: faker,
+    },
+  }),
+}
+
+// @ts-expect-error 'data' still has to be an object
+passingTesForNewDataUntypedMockedResponse.newData = () => ({
+  data: "123",
+});
+
+// Strongly typed MockedResponse is much more developer friendly
+const passingTestForNewData: MockedResponse<{ a: string }, { input: string }> = {
+  request: {
+    query: gql`query A { a }`,
+  },
+
+  newData: ({ input }) => ({
+    data: {
+      a: input
+    }
+  })
+};
+
+// @ts-expect-error returning output that doesn't match TData should fail
```

Comment:
```suggestion
// @ts-expect-error returns output that doesn't match TData
```

I think the `@ts-expect-error` is a good indicator we expect this to fail, so the "should fail" feels a bit redundant here 🙂

Reply:
done